### DISCUSSION TITLE
yarn link, multiple copies of react, developing a react plugin & testing in next

### DISCUSSION THREAD

**Discussion by rob-gordon** [vercel/next.js#20854] [Help]
I'm developing a module which exports some react components and hooks. I'd like to use `yarn link` and next.js to test out the module as I go. However, I'm getting the error seen in this image, presumably because there are multiple instances of react (the one in my yarn-link'd module and the one in next). 

<img width="1039" alt="Screen Shot 2021-01-07 at 9 39 43 AM" src="https://user-images.githubusercontent.com/159949/103905683-ec1ff580-50cc-11eb-8ccf-66923d76c838.png">

I saw that this could be [solved using a webpack alias](https://github.com/facebook/react/issues/13991#issuecomment-435587809) and that aliases could be [created in next via tsconfig.json or jsconfig.json](https://nextjs.org/docs/advanced-features/module-path-aliases) so I set up my `jsconfig.json` to look like:

```
{
  "compilerOptions": {
    "baseUrl": ".",
    "paths": {
      "react": ["node_modules/react"]
    }
  }
}
```

But I'm still getting the error. Does anyone know if I'm going about this the right way, or how it could/should be setup?

**Comment by rob-gordon** [comment]
My temporary solution right now is to symlink `my-module/node_modules/react` to `my-next-project/node_modules/react` but it seems like there is probably a less hacky way to accomplish this.

**Reply by dtbaker** [reply]
I'm doing the symlink approach now but it feels hacky, any other options?

**Reply by Nantris** [reply]
I'm not using Next.js, but this approach really simplified my life: https://github.com/facebook/react/issues/14257#issuecomment-1104331625
